src/pages/home/sidebar/SidebarScreen/index.js

```diff
@@ -0,0 +1,40 @@
+import React, {PureComponent} from 'react';
+import BaseSidebarScreen from './BaseSidebarScreen';
+
+class SidebarScreen extends PureComponent {
+    /**
+     * Method create event listener and bind.
+     * @param {BaseSidebarScreen} baseComponent
+     */
+    createDragoverListener= (baseComponent) => {
+        this.dragOverListener = this.dragOverListener.bind(baseComponent);
```

Comment:
Don't pass `baseComponent`. Instead add a ref to `BaseSidebarScreen` and use that.